### DISCUSSION TITLE
Nuget data source is not using the explicitly required query to  use semVer 2.0.

### DISCUSSION THREAD

**Discussion by juan-ibaneztorres-wd** [renovatebot/renovate#27738] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.163.0

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

So we are in the middle of onboarding some dotnet projects to use renovate. We use artifactory to host our nuget packages.
and did find that renovate was not finding the packages we currently use.
When I inspected the code I did find renovate currently uses nuget V2 and uses this code to retrieve the versions from the feed:
https://github.com/renovatebot/renovate/blob/6b065554a541303d7545da684bda70ac3cd63288/lib/modules/datasource/nuget/v2.ts#L24

Now, after further inspection I did find that this is missing a part of the query that ensures that we get semver 2.0 (renovate in theory uses it for nuget according to [this](https://docs.renovatebot.com/modules/versioning/#nuget-versioning)).

namely the url is missing a `semVerLevel=2.0.0` part to the query. as explained in [here](https://github.com/NuGet/Home/wiki/SemVer2-support-for-nuget.org-(server-side)) 

I did some testing (using curl to emulate what renovate does) and did find that as it currently stands, it cannot get all packages, but after adding the missing `semVerLevel=2.0.0` to the url then it actually retrieves the required packages.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
Previous issue: https://github.com/renovatebot/renovate/issues/15072

Much earlier PR where I appear to have added this param already: https://github.com/renovatebot/renovate/issues/3591

@viceice do you know the current status?

**Comment by juan-ibaneztorres-wd** [comment]
ahhhh sweet thanks for that. have tried the workaround using v3 and it works.
still would like to see an option to choose to use the right semVerLevel.

**Reply by viceice** [reply]
I think we had to revert that because some third-party registries didn't support that properly.

it may also cause issues with non-semver compliant versions used by some packages
